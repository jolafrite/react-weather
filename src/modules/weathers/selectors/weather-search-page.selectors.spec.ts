import {
  getSearchPageState,
  getSearchPageLoaded,
  getSearchPageLoading,
  getSearchPageError,
  getSearchPageWeatherIds,
  getSearchPageWeathers,
  getSearchPageQuery
} from "./weather-search-page.selectors";

describe('selectors: weather search page', () => {

  describe('getSearchPageState', () => {
    test('should return the weather searchPage state', () => {
      const state = {
        weathers: {
          searchPage: {
            'state': 'weathers searchPage'
          }
        }
      };

      expect(
        getSearchPageState(state as any)
      ).toMatchSnapshot();
    });
  });

  describe('getSearchPageLoaded', () => {
    test('should return if the status is loaded', () => {
      const state = {
        weathers: {
          searchPage: {
            status: 'loaded'
          }
        }
      };

      expect(
        getSearchPageLoaded(state as any)
      ).toMatchSnapshot();
    });
  });

  describe('getSearchPageLoading', () => {
    test('should return if the status is loading', () => {
      const state = {
        weathers: {
          searchPage: {
            status: 'loading'
          }
        }
      };

      expect(
        getSearchPageLoading(state as any)
      ).toMatchSnapshot();
    });
  });

  describe('getSearchPageError', () => {
    test('should return if the status is error', () => {
      const state = {
        weathers: {
          searchPage: {
            status: 'error'
          }
        }
      };

      expect(
        getSearchPageError(state as any)
      ).toMatchSnapshot();
    });
  });

  describe('getSearchPageWeatherIds', () => {
    test('should return the ids', () => {
      const state = {
        weathers: {
          searchPage: {
            ids: [1234, 4321]
          }
        }
      };

      expect(
        getSearchPageWeatherIds(state as any)
      ).toMatchSnapshot();
    });
  });

  describe('getSearchPageWeather', () => {
    test('should return the matching weathers', () => {
      const state = {
        weathers: {
          weathers: {
            entries: {
              1234: { id: 1234 },
              4321: { id: 4321 },
            }
          },
          searchPage: {
            ids: [1234, 4321]
          }
        }
      };

      expect(
        getSearchPageWeathers(state as any)
      ).toMatchSnapshot();
    });
  });

  describe('getSearchPageQuery', () => {
    test('should return the query', () => {
      const state = {
        weathers: {
          searchPage: {
            query: 'searched query'
          }
        }
      };

      expect(
        getSearchPageQuery(state as any)
      ).toMatchSnapshot();
    });
  });
});
