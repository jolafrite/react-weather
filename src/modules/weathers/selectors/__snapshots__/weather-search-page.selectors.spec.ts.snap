// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`selectors: weather search page getSearchPageError should return if the status is error 1`] = `true`;

exports[`selectors: weather search page getSearchPageLoaded should return if the status is loaded 1`] = `true`;

exports[`selectors: weather search page getSearchPageLoading should return if the status is loading 1`] = `true`;

exports[`selectors: weather search page getSearchPageQuery should return the query 1`] = `"searched query"`;

exports[`selectors: weather search page getSearchPageState should return the weather searchPage state 1`] = `
Object {
  "state": "weathers searchPage",
}
`;

exports[`selectors: weather search page getSearchPageWeather should return the matching weathers 1`] = `
Array [
  Object {
    "id": 1234,
  },
  Object {
    "id": 4321,
  },
]
`;

exports[`selectors: weather search page getSearchPageWeatherIds should return the ids 1`] = `
Array [
  1234,
  4321,
]
`;
